R/expr-string.R

```diff
@@ -107,16 +107,23 @@ expr_str_strptime <- function(
     check_dots_empty0(...)
     check_polars_dtype(dtype)
 
-    if (pl$same_outer_dt(dtype, pl$Datetime())) {
-      datetime_type <- dtype$`_rexpr`$get_insides()
-      out <- self$`_rexpr`$str_to_datetime(
-        format, datetime_type$tu, datetime_type$tz, strict, exact, cache, ambiguous
+    if (dtype$is_datetime()) {
+      datetime_type <- dtype$`_dt`$`_get_datatype_fields`()
+      time_unit <- datetime_type[["time_unit"]]
+      time_zone <- datetime_type[["time_zone"]]
```

Comment:
Why do you need to write it this way, isn't something like `time_unit` already bound to `dtype`?